dhis-2/dhis-web/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/AppHubController.java

```diff
@@ -66,20 +70,29 @@
     @Autowired
     private I18nManager i18nManager;
 
-
-    @RequestMapping( method = RequestMethod.GET, produces = "application/json" )
-    public @ResponseBody List<WebApp> listAppHub( HttpServletResponse response )
+    /**
+     * Deprecated as of version 2.35 and should be removed eventually.
+     */
+    @GetMapping( produces = "application/json" )
+    public @ResponseBody List<WebApp> listAppHub( @RequestParam String query, HttpServletResponse response )
         throws IOException
     {
         return appHubService.getAppHub();
     }
 
-    @RequestMapping( value = "/{versionId}", method = RequestMethod.POST )
+    @GetMapping( value = "/{apiVersion}", produces = "application/json" )
```

Comment:
I believe this should be `/{apiVersion}/{query}` with `query` as a `@PathVariable` instead of a `@RequestParam`, otherwise it is difficult (currently impossible) to pass query parameters to the upstream API

Reply:
Changed now.